scripts/clear-all-fault-leds.sh

```diff
@@ -64,28 +64,16 @@ done
 # they are not mapped to LED's. Applicable for CPUs and DIMMs
 if [ ${#excluded_groups} -eq 0 ]
 then
-    busctl call xyz.openbmc_project.ObjectMapper /xyz/openbmc_project/object_mapper xyz.openbmc_project.ObjectMapper \
-        GetSubTreePaths sias "/xyz/openbmc_project/inventory" 0 1 "xyz.openbmc_project.State.Decorator.OperationalStatus" \
-        | sed  's/ /\n/g' | tail -n+3 | awk -F "\"" '{print $2}' | while read -r line
+    busctl tree | grep "led/physical/" | sed 's/.*\///' | while read -r line
```

Comment:
If this script isn't going to set the functional status back, then what is?  Maybe it can do both?

Is there a defect behind this change?

Reply:
Yes, defect id 312652 is there for this. https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=312652. It seems setting of operational function loses the deconfiguration of the FRU.

Reply:
With this change, the inventory page of the web UI would show Critical health for FRUs even after the LEDs have been turned off...

Reply:
The issue here is that Gard also uses functional and we cannot touch the functional property, only way the LED can be set off as done in FSP is via setting it manually off at pre standby. The same is true for FSP based systems where the fault LEDs will be turned off on a reboot , as we are not sure if the error is still there or not , as it is from a previous boot. I do not see any issue here, Please correct me if I am missing anything

Reply:
Since we use lightpath, PELS with callouts that meet that requirements cause the PEL daemon to turn on the LEDs by setting the Functional property false.  That does 2 things - turns on the fault LED (due to the fru-fault-monitor app), and when presented in Redfish, the FRUs will have a health state of critical, which is shown in the inventory page on the web UI along with a red icon.

With the current code, this script was setting Functional back to true so that on the boot everything would also show up as healthy in addition to the LEDs being cleared.

With this change, Functional will be left as-is, so the Redfish will permanently say things are in critical health for previous errors, even when the LED is off.  If every PEL callout resulted in a guard, then it wouldn't matter, but that isn't the case.

I guess I could be wrong about this behavior, but it would be easy to test, put this PR on a system and create a PEL with a callout using busctl and see what happens on the web UI after you power off and boot again.

Reply:
I tried creating PEL with callout and see it in critical state in webUI after reboot.